### DISCUSSION TITLE
How to use CommunityToolkit in Uno

### DISCUSSION THREAD

**Discussion by DJDQfff** [unoplatform/uno#13192] [Q&A] [answered]
Followed this article: https://platform.uno/docs/articles/uno-community-toolkit.html , and with the sample code about it :https://github.com/unoplatform/Uno.Samples/tree/master/UI/UnoWCTDataGridSample. 

 I tried to use communitytoolkit in UnoPlatform.

But what I get is a confusing result.

As the article said ,use Uno. prefixed package for uno(non-windows)project and the  official for winui3 project.

My project referenced these packages like such way:

- UnoApp1: CommunityToolkit.WinUI.UI.Controls
- UnoApp1.Winodws:Uno.CommunityToolkit.WinUI.UI.Controls
- UnoApp1.Mobile: Uno.CommunityToolkit.WinUI.UI.Controls

BUt It shwo error:The Uno Platform version of Community Toolkit is not supported on WinAppSDK. Use the official packages instead

Then I checked the smaple project's packages reference way：

- WCTTest2bed:CommunityToolkit.WinUI.UI.Controls
- WCTTest2bed.Windows:CommunityToolkit.WinUI.UI.Controls
- WCTTest2bed.Windows:didn't referencr Uno.CommunityToolkit.WinUI.UI.Controls

I'm confused :
 In sample code ,the uno(non-windows)project WCTTest2bed referenced both two version of CommunityToolkit, It's not like what docs said.

Then I followed the sample code ,reference both two packages,of course ,build failed.

I have been trapped in the question for several days.

**Comment by limeniye** [comment, marked as answer]
**UnoApp1.csproj**
```XML
    <ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">
        <PackageReference Include="CommunityToolkit.WinUI.UI.Controls" Version="7.1.2"/>
    </ItemGroup>
    <ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) != 'windows'">
        <PackageReference Include="Uno.CommunityToolkit.WinUI.UI.Controls" Version="7.1.200-dev.6.gf657bb2e91"/>
    </ItemGroup>
```
![image](https://github.com/unoplatform/uno/assets/69314237/bb9be2a9-b790-465d-966e-6ffbbf2f8960)


```XML
xmlns:ui="using:CommunityToolkit.WinUI.UI"
```
